internal/pkg/service/buffer/api/service/service.go

```diff
@@ -58,28 +65,118 @@ func (*service) CreateReceiver(d dependencies.ForProjectRequest, payload *buffer
 	// Generate Secret
 	receiver.Secret = idgenerator.ReceiverSecret()
 
+	for _, exportData := range payload.Exports {
```

Comment:
Tady je asi trochu problém, je potřeba smazat ty exporty a mappingy, pokud neprojde nějaký další step. ETCD má transakce, ale nevypadá to, že by šly na tohle použít. Co s tím?

Reply:
To ide pekne vyriesit cez `Txn`. Namiesto priameho volania databazy generuj operacie, ktore naraz vykonas. Je to spolahlive a jednoduche riesenie, urcite nechces mazat co si uz vytvoril.

Mozes vsetky funkcie upravit aby generovali `Op`, aby si vedel poskladat `CreateReceiver` z viacerych `CreateExport`. Alebo mat nejaku private `createExportOp`, a public `CreateExport`.

![image](https://user-images.githubusercontent.com/19371734/203343155-c6147352-cbd3-44b3-a2cb-6a6519ed98dc.png)

To suvisi aj s tvojou otazkou ako rozdelit kod.